lib/manager/maven/index.js

```diff
@@ -0,0 +1,37 @@
+const { extractDependencies } = require('./extract');
+
+async function extractAllPackageFiles(config, packageFiles) {
+  const mavenFiles = [];
+  for (const packageFile of packageFiles) {
+    const content = await platform.getFile(packageFile);
+    if (content) {
+      const deps = extractDependencies(content);
+      if (deps) {
+        mavenFiles.push({
+          packageFile,
+          manager: 'maven',
+          ...deps,
+        });
+      } else {
+        logger.info({ packageFile }, 'can not read dependencies');
+      }
+    } else {
+      logger.info({ packageFile }, 'packageFile has no content');
+    }
+  }
+  return mavenFiles;
+}
+
+function updateDependency(fileContent, upgrade) {
+  const { currentValue, newValue, fileReplacePosition } = upgrade;
+  const leftHalf = fileContent.slice(0, fileReplacePosition);
+  const rightHalf = fileContent
+    .slice(fileReplacePosition)
+    .replace(currentValue, newValue);
```

Comment:
We need to make this "right half" logic a little smarter:

1. If the version is already updated in the right half (e.g. we are updating from 1.0.0 -> 1.0.2 and the version is already 1.0.2 before we start) then we return `fileContent` because nothing needs to change

2. If the version is the currentValue (i.e. 1.0.0) then we do the replace like above

3. If the version is neither the currentValue or the newValue (for example the pom.xml may have changed in master, or maybe it was previously updated to 1.0.1) then we should return `null`, so that the branch will be rebased off master and tried again